src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -472,19 +530,27 @@ public function dumpFile($filename, $content, $mode = 0666)
             throw new IOException(sprintf('Unable to write to the "%s" directory.', $dir), 0, null, $dir);
         }
 
-        $tmpFile = tempnam($dir, basename($filename));
+        $tmpFile = $this->tempNam($dir, basename($filename));
```

Comment:
See previous comment regarding atomicity. If dumpFile was implemented in this way, I'm not sure there would be any benefit of using it over file_put_contents?